cli/tests/integration_tests.rs

```diff
@@ -1516,6 +1516,21 @@ fn repl_test_eval_unterminated() {
   assert!(err.is_empty());
 }
 
+#[test]
```

Comment:
Validator only runs with tty so needs to be tested with pty.

Reply:
pty is not suffering from the issue. Added a test anyways.

Reply:
> pty is not suffering from the issue.

Hmm really? It should be, went through the call tree a few days ago in rustyline and the validator trait is not used at all when input is a file (e.g stdin).

Reply:
Reason is this part of the code:
https://github.com/kkawakam/rustyline/blob/e9a5ccdc4f24c4a73053e6cf8897c186da5981da/src/lib.rs#L871-L876
For pty case, rustyline calls `readline_direct` which only considers newline and does not run validator, as you mentioned:
https://github.com/kkawakam/rustyline/blob/e9a5ccdc4f24c4a73053e6cf8897c186da5981da/src/lib.rs#L757-L764